pymeasure/instruments/thyracont/smartline_v1.py

```diff
@@ -0,0 +1,144 @@
+    """Calculate a 1 bytes checksum mapped to a printable ASCII character.
+
+    The checksum is calculated by the sum of the decimal values of each message
+    character modulo 64 + 64.
+
+    :param string msg: message content
+    :returns: calculated checksum
+    :rtype: string
+    """
+    return chr(sum(map(ord, msg)) % 64 + 64)
+
+
+class SmartlineV1(Instrument):
+    """Thyracont Vacuum Instruments Smartline gauges with Communication Protocol V1.
+
+    Connection to the device is made through an RS485 serial connection.
+    The communication settings are baudrate 9600, 8 data bits, 1 stop bit,
+    no parity, no handshake.
+
+    A communication packages is structured as follows:
+
+    Characters 0-2: Address for communication
+    Character 3: Command character, uppercase letter for reading and lowercase for writing
+    Characters 4-n: Data for the command, can be empty.
+    Character n+1: Checksum calculated by: (sum of the decimal value of bytes 0-n) mod 64 + 64
+    Character n+2: Carriage return
+
+    :param adapter: pyvisa resource name of the instrument or adapter instance
+    :param string name: Name of the instrument.
+    :param int address: RS485 adddress of the instrument 1-15.
+    :param kwargs: Any valid key-word argument for Instrument
+
+    """
+
+    def __init__(self, adapter, name="Thyracont Vacuum Gauge V1", address=1, **kwargs):
+        super().__init__(adapter,
+                         name,
+                         includeSCPI=False,
+                         write_termination="\r",
+                         read_termination="\r",
+                         asrl=dict(baud_rate=9600),
+                         **kwargs)
+        self.address = address
+
+    def read(self):
+        """Reads a response message from the instrument.
+
+        This method also checks for a correct checksum.
+
+        :returns: the data fields
+        :rtype: string
+        :raises ValueError: if a checksum error is detected
+        """
+        msg = super().read()
+        chksum = calculateChecksum(msg[:-1])
+        if msg[-1] == chksum:
+            return msg[:-1]
+        else:
+            raise ValueError(
+                f"checksum error in received message {msg} "
+                f"with checksum {chksum} but received {msg[-1]}")
+
+    def write(self, command):
+        """Writes a command to the instrument.
+
+        This method adds the required address and checksum.
+
+        :param str command: command to be sent to the instrument
+        """
+        fullcmd = f"{self.address:03d}" + command
+        super().write(fullcmd + calculateChecksum(fullcmd))
+
+    def check_errors(self):
+        reply = self.read()
+        if len(reply) < 4:
+            raise ValueError(f"Reply of instrument ('{reply}') too short.")
+        if reply[3] in ["N", "X"]:
+            raise ValueError(f"Reply from Instrument indicates an error '{reply}'")
+        return []
+
+    type = Instrument.measurement(
```

Comment:
Make it better verbose:
```suggestion
    device_type = Instrument.measurement(
```